src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -165,51 +221,15 @@ struct sdm_sensor_info
       const auto slash = "/";
       const auto underscore = "_";
       /*
-       * Forming sysfs node as /<start>[start_id]_.
-       * Example: /in0_ or /curr1_ or /power1_ or /temp1_ in 1st iteration.
+       * Forming sysfs node as <path>/<start>[start_id]_. Here, path is "hwmon/hwmon*"
+       * Example: <path>/in0_ or <path>/curr1_ or <path>/power1_ or <path>/temp1_ in 1st iteration.
        * start_id will be incremented till next_id become true.
-       * So, next iteration will be /in1_ or /curr2_ or /power2_ or /temp2_.
+       * So, next iteration will be <path>/in1_ or <path>/curr2_ or <path>/power2_ or <path>/temp2_.
        * Similarly, end string of sysfs node name will be retrieved using end[].
        */
-      std::string tmp = slash + type + std::to_string(start_id) + underscore;
-      for (int end_id = 0; end_id < max_end_types; end_id++)
-      {
-        if (end_id == 0)
-        {
-          std::string errmsg;
-          std::string label;
-          pdev->sysfs_get("", path + tmp + sname_end[end_id], errmsg, label);
-          if (!errmsg.empty())
-          {
-            //go and read next sysfs node
-            data.label = "N/A";
-            next_id = true;
-            end_id = max_end_types;
-            continue;
-          }
-          data.label = label;
-        }
-        else
-        {
-          std::string errmsg;
-          uint32_t input = 0;
-          pdev->sysfs_get<uint32_t>("", path + tmp + sname_end[end_id], errmsg, input, EINVAL);
-          if (!errmsg.empty())
-            continue;
-
-          if (end_id == 1)
-            data.input = input;
-          else if (end_id == 2)
-            data.max = input;
-          else if (end_id == 3)
-            data.average = input;
-          else
-            data.highest = input;
-        }
-      } // for (end_id =0; end_id < max ...)
-
-      if (data.label.compare("N/A"))
-        output.push_back(data);
+      std::string tpath = path + slash + type + std::to_string(start_id) + underscore;
```

Comment:
unreadable, any reason `slash` and `underscore` are symbolic?  Any `type` incurs a string copy, why?